types/stylelint/index.d.ts

```diff
@@ -252,6 +253,23 @@ declare namespace stylelint {
 		fixable?: boolean;
 	};
 
+	type Range = {
+		start: {
+			line: number;
+			column: number /** inclusive */;
+		};
+		end: {
+			line: number;
+			column: number /** exclusive */;
+		};
+	};
+
+	export type FixerData = {
+		source: Range;
```

Comment:
At this point I consider this PR ready to be merged as is but I have still one last refactor in mind that might be an improvement.
As explained [here](https://github.com/stylelint/stylelint/pull/7706/files#r1615360093) there could be confusion over the terminology.
i.e. what you would expect from `end.column` to be, based on its parent name

Therefore I think renaming `source` to `range` would make sense for people who have already used a `Source` before to avoid unnecessary confusion even if the type is clear.
@ybiquitous Ill move on with the renaming shortly if you don't have anything against it.
@romainmenke does that make sense to you?